--- plle4_orig.vhd	2021-03-10 12:34:27.018447070 -0500
+++ plle4_sim.vhd	2021-03-10 15:43:37.477484637 -0500
@@ -5,2 +5,9 @@
 entity plle4_plle4_clk_wiz is
+  generic (
+    DIVCLK_DIVIDE        : integer :=  1;  -- D
+    CLKFBOUT_MULT        : integer :=  5;  -- M
+    CLKOUT0_DIVIDE       : integer :=  1;  -- O
+    CLKOUT0_PHASE        : real :=  0.0;
+    CLKOUT0_DUTY_CYCLE   : real :=  0.5;
+    CLKIN1_PERIOD        : real :=  0.0);
   port (
@@ -27,9 +34,2 @@
   attribute BOX_TYPE : string;
-  attribute BOX_TYPE of clkin1_ibuf : label is "PRIMITIVE";
-  attribute CAPACITANCE : string;
-  attribute CAPACITANCE of clkin1_ibuf : label is "DONT_CARE";
-  attribute IBUF_DELAY_VALUE : string;
-  attribute IBUF_DELAY_VALUE of clkin1_ibuf : label is "0";
-  attribute IFD_DELAY_VALUE : string;
-  attribute IFD_DELAY_VALUE of clkin1_ibuf : label is "AUTO";
   attribute BOX_TYPE of clkout1_buf : label is "PRIMITIVE";
@@ -39,10 +39,2 @@
 begin
-clkin1_ibuf: unisim.vcomponents.IBUF
-    generic map(
-      IOSTANDARD => "DEFAULT"
-    )
-        port map (
-      I => clk_in1,
-      O => clk_in1_plle4
-    );
 clkout1_buf: unisim.vcomponents.BUFGCE
@@ -58,8 +50,8 @@
     generic map(
-      CLKFBOUT_MULT => 8,
+      CLKFBOUT_MULT => CLKFBOUT_MULT,
       CLKFBOUT_PHASE => 0.000000,
-      CLKIN_PERIOD => 10.000000,
-      CLKOUT0_DIVIDE => 8,
-      CLKOUT0_DUTY_CYCLE => 0.500000,
-      CLKOUT0_PHASE => 0.000000,
+      CLKIN_PERIOD => CLKIN1_PERIOD,
+      CLKOUT0_DIVIDE => CLKOUT0_DIVIDE,
+      CLKOUT0_DUTY_CYCLE => CLKOUT0_DUTY_CYCLE,
+      CLKOUT0_PHASE => CLKOUT0_PHASE,
       CLKOUT1_DIVIDE => 1,
@@ -69,3 +61,3 @@
       COMPENSATION => "INTERNAL",
-      DIVCLK_DIVIDE => 1,
+      DIVCLK_DIVIDE => DIVCLK_DIVIDE,
       IS_CLKFBIN_INVERTED => '0',
@@ -80,3 +72,3 @@
       CLKFBOUT => NLW_plle3_adv_inst_CLKFBOUT_UNCONNECTED,
-      CLKIN => clk_in1_plle4,
+      CLKIN => clk_in1,
       CLKOUT0 => clk_out1_plle4,
@@ -104,2 +96,9 @@
 entity plle4 is
+  generic (
+    DIVCLK_DIVIDE        : integer :=  1;  -- D
+    CLKFBOUT_MULT        : integer :=  5;  -- M
+    CLKOUT0_DIVIDE       : integer :=  1;  -- O
+    CLKOUT0_PHASE        : real :=  0.0;
+    CLKOUT0_DUTY_CYCLE   : real :=  0.5;
+    CLKIN1_PERIOD        : real :=  0.0);
   port (
@@ -117,2 +116,9 @@
 inst: entity work.plle4_plle4_clk_wiz
+    generic map(
+      DIVCLK_DIVIDE        =>  DIVCLK_DIVIDE,
+      CLKFBOUT_MULT        =>  CLKFBOUT_MULT,
+      CLKOUT0_DIVIDE       =>  CLKOUT0_DIVIDE,
+      CLKOUT0_PHASE        =>  CLKOUT0_PHASE,
+      CLKOUT0_DUTY_CYCLE   =>  CLKOUT0_DUTY_CYCLE,
+      CLKIN1_PERIOD        =>  CLKIN1_PERIOD)
      port map (
